crates/generator/src/generator.rs

```diff
@@ -44,6 +45,27 @@ use ckb_vm::{
     DefaultMachineBuilder, SupportMachine,
 };
 
+struct AsmCoreMachineParams {
+    pub vm_isa: u8,
+    pub vm_version: u32,
+}
+
+impl AsmCoreMachineParams {
+    pub fn new(vm_version: u32) -> AsmCoreMachineParams {
```

Comment:
```suggestion
    pub fn with_version(vm_version: u32) -> Result<AsmCoreMachineParams> {
```

Reply:
updated to `Result<AsmCoreMachineParams, VMError>`